specs/20260421-181205-bot-workflows/data-model.md

```
@@ -0,0 +1,157 @@
+2. `UNIQUE INDEX idx_workflow_runs_inflight ON workflow_runs (workflow_name, target_owner, target_repo, target_number) WHERE status IN ('queued','running')` — enforces FR-011 idempotency: a second label application cannot open a second in-flight run for the same (workflow, item). Terminal rows do not block re-runs.
+3. `INDEX idx_workflow_runs_target ON workflow_runs (target_owner, target_repo, target_number)` — lookup on resume (`bot:ship` re-applied → "what's the latest run for this issue?").
+4. `INDEX idx_workflow_runs_parent ON workflow_runs (parent_run_id) WHERE parent_run_id IS NOT NULL` — locating children of a composite for "next step" computation.
+
+### Constraints
+
+- `CHECK ((parent_run_id IS NULL) = (parent_step_index IS NULL))` — both or neither.
+- `CHECK (target_number > 0)`.
+
+### State transitions
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)